azurelinuxagent/common/exception.py

```diff
@@ -83,21 +83,18 @@ def __init__(self, msg=None, inner=None, code=-1, exit_code=-1):
 
 
 class ExtensionUpdateError(ExtensionError):
-    """
-    When failed to update an extension
-    """
-
-    def __init__(self, msg=None, inner=None, code=-1):  # pylint: disable=W0235
-        super(ExtensionUpdateError, self).__init__(msg, inner, code)
+    # When failed to update an extension
+    pass
 
 
 class ExtensionDownloadError(ExtensionError):
-    """
-    When failed to download and setup an extension
-    """
+    # When failed to download and setup an extension
```

Comment:
Fixed 2 linter issues here - W0235 (useless-super-delegation) and W0107 (unnecessary-pass). 
The unnecessary-pass raises when we only have a pass statement with a docstring.

Reply:
Just a note: You don't technically need a `pass` statement when you have a docstring on an empty class. I.e. you could add the docstring back and remove the `pass` statement and end up with the same functionality (though the docstring will of course be added as a `__doc__` attribute, which is actually probably preferred). I believe that's what pylint is complaining about when you have the pass statement; as long as the docstring is there, the `pass` is unneccessary.

See [this stack overflow answer](https://stackoverflow.com/a/49754261).

Reply:
Ohh very cool! I didnt know about this! I'll do this instead than an empty pass if we do end up removing the super calls! Thanks!

Reply:
Yep! I actually learned about this when I got the same `unnecessary-pass` error in a prior PR, lol.